### DISCUSSION TITLE
如何修改 `'@umijs/babel-preset-umi` 的默认配置?

### DISCUSSION THREAD

**Discussion by lijinke666** [umijs/father#782] [Q&A]
https://github.com/umijs/father/blob/d647e1be8c345a73a5634413ef667436e606eb0d/src/builder/bundless/loaders/javascript/babel.ts#L90

默认配置中 `loose` 为 false, 在如下场景会有问题, 类似 https://github.com/umijs/umi/issues/9396

```
class A {
  loading = '1';
  constructor() {
  }
}
class B extends A {
  loading = '2';
}
const b = new B();
console.log(b);
```

https://github.com/umijs/umi/blob/master/packages/babel-preset-umi/src/index.ts#L31

![image](https://github.com/user-attachments/assets/409372ac-7220-4dfa-b8bc-28ae20c25e04)

在 father 这一层如何修改?

![image](https://github.com/user-attachments/assets/a2447c40-4826-4a83-9f9e-b3cd3aca063c)

**Comment by PeachScript** [comment]
暂时改不了， @Jinbao1001 看看能不能提供类似 Umi `modifyBabelPresetOpts` 的插件钩子出来，开配置项有点丑😅

**Comment by lijinke666** [comment]
同样的, esbuild swc 也有类似的问题, 如果提供钩子, 建议都考虑


eg. https://swc.rs/docs/configuration/compilation#jsctransformusedefineforclassfields

```ts
jsc: {
  transform: {
    useDefineForClassFields: false
  },
}
```

father 封装的太死了, 我还是放弃, 使用 tsc 编译算了

**Reply by PeachScript** [reply]
好的，tsc 能满足需求的话用 tsc 就好
